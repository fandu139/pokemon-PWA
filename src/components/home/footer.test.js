import React from "react";
import Render from 'react-test-renderer';
import Footer, { handleRemoveCategory, handleCategory, concatDataArray, groubArray } from './Footer';

  const dataCategoryDummy = ["Grass", "Poison", "Fire", "Flying", "Water", "Bug"]
  const dataDummy = [
    {
      "id": "UG9rZW1vbjowMDE=",
      "name": "Bulbasaur",
      "number": "001",
      "classification": "Seed PokÃ©mon",
      "types": [
        "Grass",
        "Poison"
      ],
      "resistant": [
        "Water",
        "Electric",
        "Grass",
        "Fighting",
        "Fairy"
      ],
      "attacks": {
        "fast": [
          {
            "name": "Tackle",
            "type": "Normal",
            "damage": 12
          },
          {
            "name": "Vine Whip",
            "type": "Grass",
            "damage": 7
          }
        ],
        "special": [
          {
            "name": "Power Whip",
            "type": "Grass",
            "damage": 70
          },
          {
            "name": "Seed Bomb",
            "type": "Grass",
            "damage": 40
          },
          {
            "name": "Sludge Bomb",
            "type": "Poison",
            "damage": 55
          }
        ]
      },
      "weaknesses": [
        "Fire",
        "Ice",
        "Flying",
        "Psychic"
      ],
      "fleeRate": 0.1,
      "maxCP": 951,
      "evolutions": [
        {
          "id": "UG9rZW1vbjowMDI=",
          "number": "002"
        },
        {
          "id": "UG9rZW1vbjowMDM=",
          "number": "003"
        }
      ],
      "evolutionRequirements": {
        "amount": 25,
        "name": "Bulbasaur candies"
      },
      "maxHP": 1071,
      "image": "https://img.pokemondb.net/artwork/bulbasaur.jpg"
    }
  ]

describe('Home Pokemon', () => {
  const props = {
    dataFilter: dataDummy,
    data: dataDummy,
    setData: jest.fn(),
    category: [],
    setCategory: jest.fn(),
  };

  test('Home Footer Filter By Type Pokemon Snapshoot', () => {
    const component = Render.create(<Footer {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Home Footer Remove Filter Snapshoot', () => {
    const propsManipulate = {
      ...props,
      category: dataCategoryDummy
    }
    const component = Render.create(<Footer {...propsManipulate} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Test Function HandleCategory', () => {
    const setCategory = () => {
      return true
    } 
    const result = handleCategory(dataDummy, setCategory)
    expect(result).toBe(true)
  });

  test('Test Function ConcatDataArray', () => {
    const result = concatDataArray(dataDummy)
    expect(result).toEqual([ 'Grass', 'Poison' ])
  });

  test('Test Function GroubArray', () => {
    const result = groubArray(dataCategoryDummy)
    expect(result).toEqual(dataCategoryDummy)
  });

  test('Test Function HandleRemoveCategory', () => {
    const props = { 
      setData: jest.fn(), 
      dataFilter: dataDummy, 
      setCategory: () => {
        return true
      }
    }
    const result = handleRemoveCategory(props)
    expect(result).toBe(true)
  });

});
