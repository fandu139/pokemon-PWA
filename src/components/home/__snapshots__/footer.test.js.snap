// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Pokemon Home Footer 1`] = `
<footer
  className="sc-bwzfXH fEhHCM"
  data-testid="footer-component"
>
  <div
    className="sc-bdVaJa eMKTXw sc-htpNat iBaOpL"
  >
    <div
      className="App"
    >
      <header
        className="App-button"
      >
        <div
          className="btn-toolbar"
          role="toolbar"
        >
          <button
            className="btn btn-success btn-sm"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Filter By Type Pokemon
          </button>
        </div>
      </header>
    </div>
  </div>
</footer>
`;

exports[`Home Pokemon Home Footer Filter By Type Pokemon Snapshoot 1`] = `
<footer
  className="sc-bwzfXH fEhHCM"
  data-testid="footer-component"
>
  <div
    className="sc-bdVaJa eMKTXw sc-htpNat iBaOpL"
  >
    <div
      className="App"
    >
      <header
        className="App-button"
      >
        <div
          className="btn-toolbar"
          role="toolbar"
        >
          <button
            className="btn btn-error btn-sm"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Remove Filter
          </button>
        </div>
      </header>
    </div>
  </div>
</footer>
`;

exports[`Home Pokemon Home Footer Remove Filter Snapshoot 1`] = `
<footer
  className="sc-bwzfXH fEhHCM"
  data-testid="footer-component"
>
  <div
    className="sc-bdVaJa eMKTXw sc-htpNat iBaOpL"
  >
    <div
      className="App"
    >
      <header
        className="App-button"
      >
        <div
          className="btn-toolbar"
          role="toolbar"
        >
          <button
            className="btn btn-error btn-sm"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Remove Filter
          </button>
        </div>
      </header>
    </div>
  </div>
</footer>
`;

exports[`Home Pokemon Home Footer Snapshoot 1`] = `
<footer
  className="sc-bwzfXH fEhHCM"
  data-testid="footer-component"
>
  <div
    className="sc-bdVaJa eMKTXw sc-htpNat iBaOpL"
  >
    <div
      className="App"
    >
      <header
        className="App-button"
      >
        <div
          className="btn-toolbar"
          role="toolbar"
        >
          <button
            className="btn btn-success btn-sm"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Filter By Type Pokemon
          </button>
        </div>
      </header>
    </div>
  </div>
</footer>
`;
